pkg/openstackclient/funcs.go

```diff
@@ -34,55 +40,60 @@ func ClientPod(
 		},
 	}
 
-	var terminationGracePeriodSeconds int64 = 0
-	runAsUser := int64(42401)
-	runAsGroup := int64(42401)
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["OS_CLOUD"] = env.SetValue("default")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+
 	clientPod.ObjectMeta = metav1.ObjectMeta{
 		Name:      instance.Name,
 		Namespace: instance.Namespace,
 		Labels:    labels,
 	}
-	clientPod.Spec.TerminationGracePeriodSeconds = &terminationGracePeriodSeconds
+	clientPod.Spec.TerminationGracePeriodSeconds = ptr.To[int64](0)
 	clientPod.Spec.ServiceAccountName = instance.RbacResourceName()
-	clientPod.Spec.Containers = []corev1.Container{
-		{
-			Name:    "openstackclient",
-			Image:   instance.Spec.ContainerImage,
-			Command: []string{"/bin/sleep"},
-			Args:    []string{"infinity"},
-			SecurityContext: &corev1.SecurityContext{
-				RunAsUser:  &runAsUser,
-				RunAsGroup: &runAsGroup,
+	clientContainer := corev1.Container{
+		Name:  "openstackclient",
+		Image: instance.Spec.ContainerImage,
+		Command: []string{
+			"/bin/bash",
+		},
+		Args: []string{"-c", ServiceCommand},
+		SecurityContext: &corev1.SecurityContext{
+			RunAsUser:  ptr.To[int64](42401),
+			RunAsGroup: ptr.To[int64](42401),
+		},
+		Env: env.MergeEnvs([]corev1.EnvVar{}, envVars),
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "openstack-config",
+				MountPath: "/var/lib/config-data/clouds.yaml",
+				SubPath:   "clouds.yaml",
 			},
-			Env: []corev1.EnvVar{
-				{
-					Name:  "OS_CLOUD",
-					Value: "default",
-				},
-				{
-					Name:  "CONFIG_HASH",
-					Value: configHash,
-				},
-				{
-					Name:  "SECRET_HASH",
-					Value: secretHash,
-				},
+			{
+				Name:      "openstack-config-secret",
+				MountPath: "/var/lib/config-data/secure.yaml",
+				SubPath:   "secure.yaml",
 			},
-			VolumeMounts: []corev1.VolumeMount{
-				{
-					Name:      "openstack-config",
-					MountPath: "/etc/openstack/clouds.yaml",
-					SubPath:   "clouds.yaml",
-				},
-				{
-					Name:      "openstack-config-secret",
-					MountPath: "/etc/openstack/secure.yaml",
-					SubPath:   "secure.yaml",
-				},
+			{
+				Name:      "config-data",
+				MountPath: "/var/lib/kolla/config_files/config.json",
+				SubPath:   "config.json",
+				ReadOnly:  true,
 			},
 		},
 	}
-	clientPod.Spec.Volumes = clientPodVolumes(instance, labels, configHash, secretHash)
+	if instance.Spec.CaSecretName != "" {
+		clientContainer.VolumeMounts = append(clientContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      "ca",
+				MountPath: "/var/lib/config-data/ca-certificates",
```

Comment:
Should the be ReadOnly: true?

Reply:
afaik you can not update a mounted file from a cm/secret. so its basically readonly